### PR TITLE
Iw/feature/inequality comparisons

### PR DESCRIPTION
**Pull request by IanWhitney**
TL/DR version, this implements #8 

Long version is that there's a lot of gross refactoring in the middle. Mostly around the naming of things and where certain responsibilities were. I'l try to show off the highlights.

The main problem we had to resolve was that the existing code really expected to compare two strings. Obviously with inequality operators we need to compare numbers. 

And we had to figure out how to indicate these operators in the query string. I went with a query string like "movies?q=star_rating>5". But the query string parsing expected the key and value to be separated by a "=". Meaning I had to improve the parsing **and** save that operator for later use.

We start off with documenting the new feature: ea27839
Then on to adding a feature test: f251f7f
Parsing refactoring introduced in: 44c6a30
Then there's _a lot_ of refactoring commits
In 7245745 I get back to doing real work and change Comparator so that it can accept operators.
With that I can implement the gross version of inequality comparison in 1f28de4
And in the remaining commits I refactor that gross version to a less gross version. At least we're no longer doing eval!

Because many things got refactored multiple times I recommend not commenting on individual commits. Comment on the final version of the code.

### CODE REVIEW

**Comment by IanWhitney** [comment]
Oh, blerg. Travis CI is saying it failed because Rubocop complained about a couple of tiny things. All of the tests pass.

**Comment by IanWhitney** [comment]
Man, houndci really hates it when I line up assignments over multiple lines.

File: README.md [hunk 1]
```diff
@@ -34,6 +34,20 @@ Returns every movie with a year of 1994
...
+### Return all data with values greater than or less than an attribute
+
+`movies?q=star_rating>1`
+Returns all movies with a star rating higher than one
+
+`movies?q=star_rating<3`
+Returns all movies with a star rating less than three
+
+`movies?q=star_rating>=2`
+Returns all movies with a star rating greater than or equal to 1
```
Comment (DebbieGillespie) [inline README.md]: This should read `greater than or equal to 2`

File: README.md [hunk 2]
```diff
@@ -34,6 +34,20 @@ Returns every movie with a year of 1994
 
 Returns all movies with a year of 1994 or 1995
 
+### Return all data with values greater than or less than an attribute
+
+`movies?q=star_rating>1`
+Returns all movies with a star rating higher than one
+
```
Comment (DebbieGillespie) [inline README.md:41]: I would suggest `greater than 1` so that it is consistent with the following examples.
